benchmarks/Microsoft.AspNetCore.Routing.Performance/Matchers/DfaMatcher.cs

```diff
@@ -0,0 +1,353 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace Microsoft.AspNetCore.Routing.Matchers
+{
+    internal class DfaMatcher : Matcher
```

Comment:
OK So - for folks looking at this, this is the first of two experimental matchers I have been working on. I wanted to include it in the PR at least because I've talked about it and shown some code to folks already.

What are opinions on checking in experiments in this state, **VS** waiting until something is more fleshed out? I'm happy to do whatever @kichalla and @JamesNK think is appropriate.

Reply:
We're going to a fair bit of iterating here, but I want to point out this code isn't feature complete. This can't execute some fairly basic things 😆

Reply:
I agree. Check it in!